butterknife-compiler/src/main/java/butterknife/compiler/BindingClass.java

```diff
@@ -114,14 +125,88 @@ JavaFile brewJava() {
       result.addSuperinterface(ParameterizedTypeName.get(VIEW_BINDER, TypeVariableName.get("T")));
     }
 
+    if (hasRequestedUnbinder()) {
+      result.addType(createUnbinderClass());
+    }
+
     result.addMethod(createBindMethod());
-    result.addMethod(createUnbindMethod());
 
     return JavaFile.builder(classPackage, result.build())
         .addFileComment("Generated code from Butter Knife. Do not modify!")
         .build();
   }
 
+  private TypeSpec createUnbinderClass() {
+    ClassName targetClassName = ClassName.bestGuess(targetClass);
+
+    MethodSpec unbinderConstructor = MethodSpec.constructorBuilder()
+        .addParameter(targetClassName, "target")
+        .addStatement("this.$1N = $1N", "target")
+        .build();
+
+    TypeSpec.Builder result =
+        TypeSpec.classBuilder(unbinderBinding.getUnbinderClassName().simpleName())
+        .addSuperinterface(UNBINDER)
+        .addModifiers(PRIVATE, STATIC, FINAL)
+        .addField(targetClassName, "target", PRIVATE)
+        .addMethod(unbinderConstructor);
+
+    // Even if there are no bindings we need to implement the interface method.
+    MethodSpec.Builder unbindMethod = MethodSpec.methodBuilder("unbind")
```

Comment:
Yes but i'd rather the exception be explicit that it was the user's fault. Otherwise I'll get an issue filed in a few weeks saying that BK cause an NPE.